File: src/test/java/org/veupathdb/lib/blast/BlastFormatterTest.java
package org.veupathdb.lib.blast;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;
import org.veupathdb.lib.blast.consts.Flag;

import static org.junit.jupiter.api.Assertions.*;

@DisplayName("BlastFormatter")
class BlastFormatterTest
{
  private final ObjectMapper json = new ObjectMapper();

  @Nested
  @DisplayName(Flag.RequestID)
  class RequestID
  {
    @Test
    @DisplayName("serialization")
    void test1() throws Exception {
      var tgt = new BlastFormatter();
      tgt.setRequestID("hi");

      assertEquals("{\"-rid\":\"hi\"}", json.writeValueAsString(tgt));
    }

    @Test
    @DisplayName("deserialization")
    void test2() throws Exception {
      var raw = "{\"-rid\":\"hi\"}";
      var tgt = json.readValue(raw, BlastFormatter.class);

      assertEquals("hi", tgt.getRequestID());
    }
  }

  @Nested
  @DisplayName(Flag.ArchiveFile)
  class ArchiveFile
  {
    @Test
    @DisplayName("serialization")
    void test1() throws Exception {
      var tgt = new BlastFormatter();
      tgt.setArchiveFile("hi");

      assertEquals("{\"-archive\":\"hi\"}", json.writeValueAsString(tgt));
    }

    @Test
    @DisplayName("deserialization")
    void test2() throws Exception {
      var raw = "{\"-archive\":\"hi\"}";
      var tgt = json.readValue(raw, BlastFormatter.class);

      assertEquals("hi", tgt.getArchiveFile());
    }
  }
}

File: src/test/java/org/veupathdb/lib/blast/BlastFormatterTest.kt
package org.veupathdb.lib.blast

import com.fasterxml.jackson.databind.ObjectMapper
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Nested
import org.junit.jupiter.api.Test
import org.veupathdb.lib.blast.consts.Flag

@DisplayName("BlastFormatter")
internal class BlastFormatterTest {
  private val json = ObjectMapper()

  @Nested
  @DisplayName(Flag.RequestID)
  internal inner class RequestID {
    @Test
    @DisplayName("serialization")
    @Throws(Exception::class)
    fun test1() {
      val tgt = BlastFormatter()
      tgt.requestID = "hi"
      Assertions.assertEquals("{\"-rid\":\"hi\"}", json.writeValueAsString(tgt))
    }

    @Test
    @DisplayName("deserialization")
    @Throws(Exception::class)
    fun test2() {
      val raw = "{\"-rid\":\"hi\"}"
      val tgt = json.readValue(raw, BlastFormatter::class.java)
      Assertions.assertEquals("hi", tgt.requestID)
    }
  }

  @Nested
  @DisplayName(Flag.ArchiveFile)
  internal inner class ArchiveFile {
    @Test
    @DisplayName("serialization")
    @Throws(Exception::class)
    fun test1() {
      val tgt = BlastFormatter()
      tgt.archiveFile = "hi"
      Assertions.assertEquals("{\"-archive\":\"hi\"}", json.writeValueAsString(tgt))
    }

    @Test
    @DisplayName("deserialization")
    @Throws(Exception::class)
    fun test2() {
      val raw = "{\"-archive\":\"hi\"}"
      val tgt = json.readValue(raw, BlastFormatter::class.java)
      Assertions.assertEquals("hi", tgt.archiveFile)
    }
  }
}